logs/stream/stream.go

```diff
@@ -169,17 +208,17 @@ func AnalyzeStreamInGroups(logLines []state.LogLine) (state.TransientLogState, s
 		return state.TransientLogState{}, state.LogFile{}, tooFreshLogLines, nil
 	}
 
-	logFile, err := writeTmpLogFile(readyLogLines)
-	if err != nil {
-		return state.TransientLogState{}, state.LogFile{}, logLines, err
-	}
-
 	// Ensure that log lines that span multiple lines are already concated together before passing them to analyze
 	//
 	// Since we already sorted by PID earlier, it is safe for us to concatenate lines before grouping. In fact,
 	// this is required for cases where unknown log lines don't have PIDs associated
 	analyzableLogLines := stitchLogLines(readyLogLines)
 
+	logFile, err := writeTmpLogFile(analyzableLogLines)
```

Comment:
Is the `tmpLogFile` the log equivalent of persistent state?

Reply:
It's a temporary file thats used to encrypt the log data before transmission - it's easier to do so on disk, vs in memory. The file is deleted at the end of each upload, i.e. does not persist between calls of this function.